ext_mongodb.php

```diff
@@ -227,6 +333,57 @@ public function __wakeup() : void;
 
 	<<__Native>>
 	public function selectServer(ReadPreference $readPreference): Server;
+
+/* {{{ APM */
+	public function addSubscriber( Monitoring\Subscriber $subscriber ) : void
+	{
+		$this->subscribers[] = $subscriber;
```

Comment:
An easy fix would be to copy the `foreach` from `removeSubscriber()` and have it return and NOOP if it finds the subscriber already in the array. That will let you keep the array count check in the C++ code as-is without migrating to SplObjectStorage. Seems like a small amount of effort to avoid a weird edge case.

This does make `addSubscriber()` O(n), but it's not a critical function.